src/main/java/com/google/devtools/build/lib/cmdline/RepositoryMapping.java

```diff
@@ -93,4 +95,17 @@ public RepositoryName get(String preMappingName) {
       return RepositoryName.createUnvalidated(preMappingName).toNonVisible(ownerRepo());
     }
   }
+
+  public Optional<String> getLocalNameFromMainRepo(RepositoryName postMappingName) {
+    if (ownerRepo() == null) {
+      // This can only happen if Bzlmod is not enabled. In this case, assume no repo mappings are
+      // in use, so that canonical and  local names can be used interchangeably.
+      return Optional.of(postMappingName.getName());
+    }
+    Preconditions.checkArgument(ownerRepo().isMain());
+    return repositoryMapping().entrySet().stream()
+        .filter(e -> e.getValue().equals(postMappingName))
+        .map(Entry::getKey)
+        .findFirst();
```

Comment:
This is `O(num mappings)`, so a proper implementation would instead store the mapping in an `ImmutableSetMultimap`.